drivers/power/adp1050/iio_adp1050.c

```diff
@@ -0,0 +1,878 @@
+	int ret, i;
+	uint8_t val;
+
+	switch (iio_adp1050->chan) {
+	case ADP1050_OUTA:
+	case ADP1050_OUTB:
+		ret = adp1050_read(adp1050, ADP1050_OUTA_OUTB_MODULATION_SETTINGS, &val, 1);
+		if (ret)
+			return ret;
+
+		break;
+	case ADP1050_SR1:
+	case ADP1050_SR2:
+		ret = adp1050_read(adp1050, ADP1050_SR1_SR2_MODULATION_SETTINGS, &val, 1);
+		if (ret)
+			return ret;
+
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	for (i = 0; i < ADP1050_MODULATION_AVAIL_SIZE; i++)
+		if (no_os_field_get(ADP1050_MODULATION_AVAIL_MASK(i), val))
+			return sprintf(buf, "%s ",
+				       adp1050_modulation_avail[ADP1050_MODULATION_AVAIL_MASK(i)]);
+
+	if (i == ADP1050_OUTPUT_CHANNELS)
+		return -EINVAL;
+}
+
+static int adp1050_iio_read_modulation_available(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv)
+{
+	int length = 0;
+	int i;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(adp1050_modulation_avail); i++)
+		length += sprintf(buf + length, "%s ", adp1050_modulation_avail[i]);
+
+	return length;
+}
+
+static int adp1050_iio_write_modulation(void *dev, char *buf, uint32_t len,
+					const struct iio_ch_info *channel,
+					intptr_t priv)
+{
+	struct adp1050_iio_desc *iio_adp1050 = dev;
+	struct adp1050_desc *adp1050 = iio_adp1050->adp1050_desc;
+	uint32_t i;
+
+	for (i = 0; i < ADP1050_OUTPUT_CHANNELS; i++)
+		if (!strcmp(buf, adp1050_modulation_avail[ADP1050_MODULATION_AVAIL_MASK(i)]))
+			break;
+
+	if (i == ADP1050_OUTPUT_CHANNELS)
+		return -EINVAL;
+
+	return adp1050_pwm_modulation(adp1050,
+				      (enum adp1050_mod)ADP1050_MODULATION_AVAIL_SELECT(i), iio_adp1050->chan,
+				      ADP1050_MODULATION_AVAIL_MASK(i) % 2);
+}
+
+static int adp1050_iio_read_duty_cycle(void *dev, char *buf, uint32_t len,
+				       const struct iio_ch_info *channel,
+				       intptr_t priv)
+{
+
+}
+
+static int adp1050_iio_write_duty_cycle(void *dev, char *buf, uint32_t len,
+					const struct iio_ch_info *channel,
+					intptr_t priv)
+{
+
+}
+
+static int adp1050_iio_read_enable(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+
+}
+
+static int adp1050_iio_read_enable_available(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+
+}
+
+static int adp1050_iio_write_enable(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+
+}
+
+static int adp1050_iio_update_channels(void *dev, uint32_t mask)
+{
+
+}
+
+static int adp1050_iio_reg_read(struct adp1050_iio_desc *dev, uint16_t reg,
+				uint16_t *readval, uint8_t bytes_number)
+{
+
+}
+
+static int adp1050_iio_reg_write(struct adp1050_iio_desc *dev, uint16_t reg,
+				 uint16_t writeval, uint8_t bytes_number)
+{
+
+}
+
+int adp1050_iio_init(struct adp1050_iio_desc **,
+		     struct adp1050_iio_desc_init_param *)
+{
+
+}
+
+int adp1050_iio_remove(struct adp1050_iio_desc *)
+{
+
+}
```

Comment:
these are not implemented, maybe just leave the PR as draft if you are still performing changes to it which are not related to the reviewing process

Reply:
Indeed the iio driver is still in the works so I will mark the PR as a draft.